src/Services/DataService/DataService.test.ts

```diff
@@ -334,6 +346,58 @@ describe("DataService", () => {
             expect(pushAllAction.calledWithMatch(expectedResult)).to.be.true;
           });
     });
+
+    it("should call serializeQueryParams with the queryParams from the payload", () => {
+      const serializeQueryParamsStub = 
+        stub(fakeService.serializer, "serializeQueryParams");
+
+      fakeService.fetchAllEpic(ActionsObservable.of(fetchAllAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(serializeQueryParamsStub.firstCall.args[0]).to.equal(queryParams);
+          });      
+    });
+
+    it("should call normalizeQueryResponse with the fetchAll response", () => {
+      const items = [];
+      const response = {...createMockQueryResponse(), items} as IRawQueryResponse<any>;
+      mockAdapter.fetchAll.returns(response);
+      const normalizeQueryResponseStub = stub(fakeService.mapper, "normalizeQueryResponse");
+
+      fakeService.fetchAllEpic(ActionsObservable.of(fetchAllAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(normalizeQueryResponseStub.firstCall.args[0]).to.equal(response);
+          }); 
+    });
+
+    it("should call setQueryResult after pushAll", () => {
+      const items = [];
+      const response = {...createMockQueryResponse(), items} as IRawQueryResponse<any>;
+      stub(fakeService.mapper, "normalizeQueryResponse").returns(response);
+      const pushAllStub = stub(fakeService.actions, "pushAll");
+      const setQueryResponseStub = stub(fakeService.actions, "setQueryResponse");
+      fakeService.fetchAllEpic(ActionsObservable.of(fetchAllAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(setQueryResponseStub.calledImmediatelyAfter(pushAllStub)).to.be.true;
+          }); 
+    });
+
+    it("should call setQueryResult with the response from the pushAll action", () => {
```

Comment:
This is not a valid test. setQueryResult should be called with an IQueryResponse and the IQueryBuilder it was given. As I mentioned last week, we may need to update the epic being tested as I was concerned that it was not set up properly.

Reply:
I fixed this test and updated the fetchAllEpic. It should be all set now 👍